### PR TITLE
Fix `aten.masked_fill.Scalar` and `aten.masked_fill.Tensor` lowering

### PR DESCRIPTION
**Pull request by pschang-phy**
### Ticket
https://github.com/tenstorrent/pytorch2.0_ttnn/issues/483

### Problem description
`aten.masked_fill.Scalar` and `aten.masked_fill.Tensor` fail to lowering into TTNN ops.

### What's changed
1. Fix the `ttnn.full` usage, the fill value should be set in `kwargs["fill_value"]`
2. extract fill value from `getattr` constant in `aten.masked_fill.Tensor` argument 

Test
```bash
pytest -s tests/lowering/creation/test_masked_fill.py
```

### CODE REVIEW

**Review by ayerofieiev-tt — APPROVED** [review]

File: torch_ttnn/passes/lowering/to_tt_pass.py [hunk 1]
```diff
@@ -828,17 +829,28 @@ def batch_norm_inference(input, weight, bias, mean, var, momentum, eps):
...
                 np_mask_shp = np.array(mask_shape)
                 if np.sum(np_tensor_shp % np_mask_shp) == 0:
-                    multiplier = np_tensor_shp // np_mask_shp
-                    mask_bcst = g.call_function(target_wrappers.repeat, args=(mask, multiplier.tolist()))
+                    if isinstance(fill_value, (float,)):
+                        fill_real_value = fill_value
+                    elif fill_value.op == "get_attr":
+                        with unset_fake_temporarily():
+                            fill_value_attr = getattr(gm, fill_value.target)
+                            fill_real_value = fill_value_attr.item()
```
Comment (jerrysky3) [inline torch_ttnn/passes/lowering/to_tt_pass.py:837]: Had an offline discussion. This is to handle the special case when `masked_fill.Tensor` is called with a 0-D fill value tensor. Maybe we can add some comments here and guard unsupported cases

**Comment by swimdi** [comment]
There have `aten_masked_fill_scalar_blocklist` in `to_tt_guard.py` & `to_tt_guard_autogen.py` I think you could remove

File: torch_ttnn/passes/lowering/to_tt_pass.py (hunk 1 above)
Comment (swimdi) [reply torch_ttnn/passes/lowering/to_tt_pass.py:837]: I think the unsupported cases should fire an issue for each so that can show on the roadmap

Comment (pschang-phy) [reply torch_ttnn/passes/lowering/to_tt_pass.py]: I have revised to more general condition check 54f20ca107a3119627d471ce5d399d7105444f2a for all case of constant, support either scalar or `torch.tensor`. Also add comment to mention this.

For all our current tests, not encounter unsupported cases. Furthermore, it seems in `torch.masked_fill` API not support tensor fill value (only support scalar) directly, wondering how this **distilbert-base-uncased** model got this `masked_fill.Tensor` op

File: tests/lowering/creation/test_masked_fill.py
Comment (jerrysky3) [inline tests/lowering/creation/test_masked_fill.py:1]: Maybe we can also add test cases for the current unsupported cases and make sure they fallback correctly

**Review by jerrysky3 — APPROVED** [review]

File: tests/lowering/creation/test_masked_fill.py
Comment (pschang-phy) [reply tests/lowering/creation/test_masked_fill.py:1]: it seems in `torch.masked_fill` API not support input with **tensor fill value** (only support scalar) directly.

**Pull request merged**